File: services/auth_service/internal/service/user_service.go
package service

import (
	"auth_service/internal/dto"
	"auth_service/internal/models"
	"auth_service/internal/repository"
)

type UserService struct {
	userRepo repository.UserRepository
}

func NewUserService(userRepo repository.UserRepository) *UserService {
	return &UserService{userRepo: userRepo}
}

func (s *UserService) CreateUser(req *dto.CreateUserRequest) (*models.User, error) {
	existingUser, _ := s.userRepo.Get(req.TelegramID)
	if existingUser != nil {
		return existingUser, nil
	}

	user := models.User{
		TelegramID: req.TelegramID,
		Username: req.Username,
		FullName: req.FullName,
	}

	return s.userRepo.Create(&user)
}

func (s *UserService) GetUser(telegramID int64) (*models.User, error) {
	return s.userRepo.Get(telegramID)
}

func (s *UserService) UpdateUser(user *models.User) error {
	return s.userRepo.Update(user)
}

func (s *UserService) UpdateUserRoles(user *models.User) error {
	var roles []string
	for _, role := range user.Roles {
		roles = append(roles, role.Name)
	}
	return s.userRepo.UpdateUserRoles(user.TelegramID, roles)
}

func (s *UserService) DeleteUser(user *models.User) error {
	return s.userRepo.Delete(user.TelegramID)
}

func(s *UserService) ListUsers() ([]models.User, error) {
	return s.userRepo.List()
}

File: services/bot_service/app/services/user_service.py
from typing import Optional
from app.services.auth_service import auth_service
from app.models import UserResponse, TelegramUser


class UserService:
    """
    Класс для удобного взаимодействия с пользователями
    """
    async def register_user(self, telegram_user: TelegramUser) -> Optional[UserResponse]:
        """Регистрирует пользователя в системе"""
        user = await auth_service.create_user(
            telegram_id=telegram_user.id,
            username=telegram_user.username,
            full_name=telegram_user.full_name
        )
        return user

    async def get_user(self, user_id: int) -> Optional[UserResponse]:
        user = await auth_service.get_user(user_id)
        return user


user_service = UserService()
